src/domain/update-developer-issue.use-case.ts

```diff
@@ -0,0 +1,13 @@
+import { Service } from 'typedi';
+import { DeveloperDataSource } from '@data/db';
+
+@Service()
+export class UpdateDeveloperIssueUseCase {
+  constructor(private readonly dataSource: DeveloperDataSource) { }
+
+  async execute(developerId: number, issueId: number): Promise<boolean> {
+    const success = await this.dataSource.update(developerId, { issueId });
+
+    return Promise.resolve(success);
```

Comment:
Aqui você pode tirar o async e o await e retornar direto o `this.dataSource.update(developerId, { issueId })`